<!--  -->
<launch>

  <!-- ========= Bagplayer ========== -->
  <arg name="bagname" default="/home/razlaw/bags/obstacle_detection_dji/velodyne_dji_016-11-08-16-57-12_object_detection.bag"/>
  <arg name="rate" default="0.1"/>
  <arg name="start" default="0"/>

  <node pkg="rosbag" type="play" name="play" args="--clock -r $(arg rate) -s $(arg start) $(arg bagname)" output="screen" respawn="false" >
    <remap from="/velodyne_points" to="/velodyne_points_frombag" />
  </node>


  <!-- ============= Rviz / rqt =============  -->
  <arg name="use_rqt" default="true"/>
  <node unless="$(arg use_rqt)" name="rivz" pkg="rviz" type="rviz" />
  <node if="$(arg use_rqt)" name="rqt_gui" pkg="rqt_gui" type="rqt_gui" output="screen" required="true" args=""/>


  <!-- ========= Config Server ========== -->
  <!-- will try to read an initial configuration file form CONFIGPATH/config_ROBOTNAME -->
  <node if="$(arg use_rqt)" name="config_server" pkg="config_server" type="config_server" output="screen" required="true">
    <param name="config_path" value="CONFIGPATH" />
    <param name="robot_name" value="ROBOTNAME" />
    <param name="robot_type" value="ROBOTTYPE" />
  </node>


  <!-- ============= Pointcloud Conversion (incl. nodelet_manager) =============  -->
  <arg name="frame_id" default="velodyne" />
  <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch">
    <arg name="frame_id" value="$(arg frame_id)"/>
    <arg name="min_range" value="0.01" />
  </include>


  <!-- ============= Object Detection =============  -->
  <arg name="manager" default="velodyne_nodelet_manager" />
  <arg name="points_topic" default="/velodyne_points" />
  <arg name="certainty_threshold" default="0.0" />
  <arg name="median_small_kernel_size" default="9" />
  <arg name="median_big_kernel_size" default="41" />
  <arg name="distance_to_comparison_points" default="3" />
  <arg name="publish_filtered_cloud" default="false" />

  <include file="$(find velodyne_object_detector)/launch/object_detector_nodelet.launch">
    <arg name="manager" value="$(arg manager)" />
    <arg name="points_topic" value="$(arg points_topic)" />
    <arg name="certainty_threshold" value="$(arg certainty_threshold)" />
    <arg name="median_small_kernel_size" value="$(arg median_small_kernel_size)" />
    <arg name="median_big_kernel_size" value="$(arg median_big_kernel_size)" />
    <arg name="distance_to_comparison_points" value="$(arg distance_to_comparison_points)" />
    <arg name="publish_filtered_cloud" value="$(arg publish_filtered_cloud)" />
  </include>

</launch>
